src/plugins/saved_objects_management/public/management_section/objects_table/components/duplicate_modal.tsx

```diff
@@ -0,0 +1,467 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React from 'react';
+import { FormattedMessage } from '@osd/i18n/react';
+import { groupBy } from 'lodash';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+  EuiComboBox,
+  EuiFormRow,
+  EuiCheckbox,
+  EuiComboBoxOptionOption,
+  EuiInMemoryTable,
+  EuiToolTip,
+  EuiIcon,
+  EuiCallOut,
+  EuiText,
+} from '@elastic/eui';
+import {
+  HttpSetup,
+  NotificationsStart,
+  WorkspaceAttribute,
+  WorkspacesStart,
+} from 'opensearch-dashboards/public';
+import { i18n } from '@osd/i18n';
+import { SavedObjectWithMetadata } from '../../../../common';
+import { getSavedObjectLabel } from '../../../../public';
+
+type WorkspaceOption = EuiComboBoxOptionOption<WorkspaceAttribute>;
+
+export enum DuplicateMode {
+  Selected = 'selected',
+  All = 'all',
+}
+export interface ShowDuplicateModalProps {
+  onDuplicate: (
+    savedObjects: SavedObjectWithMetadata[],
+    includeReferencesDeep: boolean,
+    targetWorkspace: string
+  ) => Promise<void>;
+  http: HttpSetup;
+  workspaces: WorkspacesStart;
+  duplicateMode: DuplicateMode;
+  notifications: NotificationsStart;
+  selectedSavedObjects: SavedObjectWithMetadata[];
+}
+
+interface Props extends ShowDuplicateModalProps {
+  onClose: () => void;
+}
+
+interface State {
+  allSelectedObjects: SavedObjectWithMetadata[];
+  workspaceOptions: WorkspaceOption[];
+  allWorkspaceOptions: WorkspaceOption[];
+  targetWorkspaceOption: WorkspaceOption[];
+  isLoading: boolean;
+  isIncludeReferencesDeepChecked: boolean;
+  savedObjectTypeInfoMap: Map<string, [number, boolean]>;
+}
+
+function capitalizeFirstLetter(str: string): string {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+}
+
+export class SavedObjectsDuplicateModal extends React.Component<Props, State> {
+  private isMounted = false;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      allSelectedObjects: this.props.selectedSavedObjects,
+      workspaceOptions: [],
+      allWorkspaceOptions: [],
+      targetWorkspaceOption: [],
+      isLoading: false,
+      isIncludeReferencesDeepChecked: true,
+      savedObjectTypeInfoMap: new Map<string, [number, boolean]>(),
+    };
+  }
+
+  workspaceToOption = (
+    workspace: WorkspaceAttribute,
+    currentWorkspaceName?: string
+  ): WorkspaceOption => {
+    // add (current) after current workspace name
+    let workspaceName = workspace.name;
+    if (workspace.name === currentWorkspaceName) {
```

Comment:
Better to use id to check if it's current workspace

Reply:
Yes, it had been modified